openedx/core/djangoapps/programs/utils.py

```diff
@@ -142,6 +179,11 @@ def progress(self, programs=None, count_only=True):
             list of dict, each containing information about a user's progress
                 towards completing a program.
         """
+        now = datetime.datetime.now(utc)
+        enrolled_run_modes = defaultdict(set)
```

Comment:
Why are you using a set here? I don't think it's possible to be enrolled in more than one mode for a given course run.